server/src/main/java/com/paypal/selion/grid/matchers/SeLionSauceCapabilityMatcher.java

```diff
@@ -17,19 +17,54 @@
 
 import java.util.Map;
 
+import org.apache.commons.lang.StringUtils;
 import org.openqa.grid.internal.utils.CapabilityMatcher;
-import org.openqa.grid.internal.utils.DefaultCapabilityMatcher;
+
+import com.paypal.selion.proxy.SeLionSauceProxy;
 
 /**
- * This is a {@link CapabilityMatcher} based extension of {@link DefaultCapabilityMatcher} that nullifies the 
- * capabilities of the internal {@link DefaultCapabilityMatcher} which resides in the Grid.
- * This is essentially done to ensure that all of the {@link CapabilityMatcher} capabilities are delegated
- * to the one that dwells in the Sauce Labs cloud.
+ * Used in conjunction with a grid that has a {@link SeLionSauceProxy} connected. <br>
+ * <br>
+ * A simple {@link CapabilityMatcher} which checks for required information in the requested desired capabilities but
+ * does not perform any match against node capabilities. Matches against node capabilities will be performed by the
+ * Sauce labs cloud when {@link SeLionSauceProxy} forwards the session.
  * 
  */
-public class SeLionSauceCapabilityMatcher extends DefaultCapabilityMatcher {
-    @Override
-    public boolean matches(Map<String, Object> nodeCapability, Map<String, Object> requestedCapability) {
+public class SeLionSauceCapabilityMatcher implements CapabilityMatcher {
+    private static final String SAUCE_USER_NAME_CAPABILITY = "sauceUserName";
+    private static final String SAUCE_API_KEY_CAPABILITY = "sauceApiKey";
+    private static final String ACCESS_KEY_CAPABILITY = "accessKey";
+    private static final String USERNAME_CAPABILITY = "username";
+
+    public boolean matches(Map<String, Object> nodeCapability, Map<String, Object> requestedCapabilities) {
+        // check for backwards compatible capability names that might come across the wire
+        // TODO remove backwards compatible support in a future version.
+        processDeprecatedCapabilities(requestedCapabilities);
+
+        // refuse the session when there is no sauce user name and/or access key
+        if (isMissingRequiredCapabilities(requestedCapabilities)) {
+            // insert a hint in the requested capabilities map
+            requestedCapabilities.put("invalid-sauce-credentials", "true");
+            return false;
+        }
         return true;
     }
+
+    private void processDeprecatedCapabilities(Map<String, Object> requestedCapabilities) {
+        final String username = (String) requestedCapabilities.get(SAUCE_USER_NAME_CAPABILITY);
+        if (StringUtils.isNotEmpty(username)) {
```

Comment:
Should we put a print out warning saying capability key sauceUserName is deprecated, please use username